### PR TITLE
use service account creds to talk to cromwell

### PR DESCRIPTION
**Pull request by dvoet**
https://broadinstitute.atlassian.net/browse/GAWB-618
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Make sure liquibase is updated if appropriate
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [x] **Assign to submitter** to finalize
- [x] **Submitter**: Squash commits, rebase if necessary
- [x] **Submitter**: Verify all tests go green, including CI tests
- [x] **Submitter**: Merge to develop 
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -42,7 +42,9 @@ libraryDependencies ++= {
     "org.broadinstitute.dsde.vault" %% "vault-common" % "0.1-15-bf74315",
     ("com.google.apis" % "google-api-services-storage" % "v1-rev30-1.20.0").exclude("com.google.guava", "guava-jdk5"),
     ("com.google.apis" % "google-api-services-compute" % "v1-rev72-1.20.0"),
+    ("com.google.apis" % "google-api-services-plus" % "v1-rev381-1.20.0"),
```
Comment (dvoet) [inline build.sbt:45]: needed for Plus oauth scopes (email and profile)

File: build.sbt [hunk 2]
```diff
@@ -42,7 +42,9 @@ libraryDependencies ++= {
...
     ("com.google.apis" % "google-api-services-admin-directory" % "directory_v1-rev53-1.20.0"),
+    "com.google.apis" % "google-api-services-oauth2" % "v1-rev112-1.20.0",
```
Comment (dvoet) [inline build.sbt:47]: needed for the TokenInfo api call

File: src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 3]
```diff
@@ -72,6 +75,8 @@ object Boot extends App {
       ldapConfig.getString("userDnFormat")
     )
 
+    enableServiceAccount(gcsDAO, userDirDAO)
+
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:79]: automatically enable in ldap the service account user so we don't need to remember to do it in each environment

File: src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala [hunk 4]
```diff
@@ -123,5 +128,24 @@ object Boot extends App {
...
+   */
+  def enableServiceAccount(gcsDAO: HttpGoogleServicesDAO, userDirDAO: JndiUserDirectoryDAO): Unit = {
+    val enableServiceAccountFuture = for {
+      serviceAccountUser <- gcsDAO.getServiceAccountRawlsUser()
+      _ <- userDirDAO.createUser(serviceAccountUser).recover { case e: NameAlreadyBoundException => Unit } // if it already exists, ok
+      _ <- userDirDAO.enableUser(serviceAccountUser).recover { case e: AttributeInUseException => Unit } // if it is already enabled, ok
+    } yield Unit
+
+    enableServiceAccountFuture.onFailure {
+      // this is logged as a warning because almost always the service account is already enabled
```
Comment (jmthibault79) [inline src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:144]: I'm not sure I understand this comment.  This is a real error, so why not log it as such?

**Comment by jmthibault79** [comment]
Mostly 👍 with one comment

File: src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala (hunk 4 above)
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/Boot.scala:144]: judgement call really... if an error happens here, the only time it really matters _at this point in the code_ is when a new service account is in use.

**Comment by helgridly** [comment]
@dvoet did you ping me on hipchat because you want me to LR this, or just so I know it's incoming?

**Pull request merged**